presto-main/src/test/java/com/facebook/presto/cost/TestFilterStatsCalculator.java

```diff
@@ -244,4 +246,49 @@ public void testNotStats()
                                 .distinctValuesCount(20.0)
                                 .nullsFraction(0.4)); // FIXME - nulls shouldn't be restored
     }
+
+    @Test
+    public void testIsNullFilter()
+    {
+        Expression isNullPredicate = new IsNullPredicate(new SymbolReference("x"));
+        assertExpression(isNullPredicate)
+                .outputRowsCount(250.0)
+                .symbolStats(new Symbol("x"), symbolStats -> {
+                    symbolStats.distinctValuesCount(0)
+                            .emptyRange()
+                            .nullsFraction(1.0);
+                });
+
+        Expression isNullEmptyRangePredicate = new IsNullPredicate(new SymbolReference("emptyRange"));
+        assertExpression(isNullEmptyRangePredicate)
+                .outputRowsCount(1000.0)
+                .symbolStats(new Symbol("emptyRange"), symbolStats -> {
+                    symbolStats.distinctValuesCount(0.0)
+                            .emptyRange()
+                            .nullsFraction(1.0);
+                });
+    }
+
+    @Test
+    public void testIsNotNullFilter()
+    {
+        Expression isNotNullPredicate = new IsNotNullPredicate(new SymbolReference("x"));
+        assertExpression(isNotNullPredicate)
+                .outputRowsCount(750.0)
+                .symbolStats("x", symbolStats -> {
+                    symbolStats.distinctValuesCount(40.0)
+                            .lowValue(-10.0)
+                            .highValue(10.0)
+                            .nullsFraction(0.0);
+                });
+
+        Expression isNotNullEmptyRangePredicate = new IsNotNullPredicate(new SymbolReference("emptyRange"));
+        assertExpression(isNotNullEmptyRangePredicate)
+                .outputRowsCount(0.0)
+                .symbolStats("emptyRange", symbolStats -> {
+                    symbolStats.distinctValuesCount(0.0)
+                            .emptyRange()
+                            .nullsFraction(1.0);
+                });
+    }
```

Comment:
add a test to `TestTpchLocalStats`

Reply:
In TPCH all columns are effectively non-null.